plugins/generator-1.20.1/forge-1.20.1/templates/block/block.java.ftl

```diff
@@ -319,13 +341,26 @@ public class ${name}Block extends
 	    	    .setValue(FACE, AttachFace.WALL)
 	    	</#if>
 	    	.setValue(FACING, context.getClickedFace())
+	    	<@initCustomBlockStateProperties />
 	    	<#if data.isWaterloggable>
 	    	.setValue(WATERLOGGED, flag)
 	    	</#if>;
 		</#if>
 	}
 	</#if>
 
+	<#macro initCustomBlockStateProperties>
+		<#list data.customProperties as prop>
+			.setValue(${prop.property.getName().replace("CUSTOM:", "")?upper_case},

```

Comment:
This is a good idea, but I initially intended default values to be determined automatically, namely the first of possible values, as this seems to be the case with vanilla properties.

Reply:
Hmm, but if you would use PropertyDataWithValue later, this would require a converter.

I don't think it is wise to not let one specify default values, as it is logical one may expect something outside default values possible.

Maybe it may make sense to consider this in this PR already

Reply:
I wasn't sure how this would look like, right now the panel has 3 sections and I already had to figure how to force them to not take too much space in the scroll pane (with states list included, they could get extended behind that second list). Maybe if default/allowed values take another row on the property panel?

Reply:
I still see quite a lot of space here for default value:

![image](https://github.com/user-attachments/assets/583ca716-e40b-43dc-8a23-d4c6a2bdfcc5)

Reply:
Imagine this list 2 times narrower when states are added. I'll try with another row.

Reply:
Well, you can change it later. Don't believe other PRs will make it in one MCreator so making UI weird now is a no-go. Two rows could be quite cluttered.

You could also later add states in NORTH-SOUTH instead of WEST-EAST layout

Reply:
I can't see how that would look because states themselves will be a lot of items each with a lot of parameters (at some point).

Reply:
> For user it will be quite strange having everything cluttered currently in rows and what not while having this huge space on the right side.

> Well, you can change it later